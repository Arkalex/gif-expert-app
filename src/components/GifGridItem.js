import React from 'react'

export const GifGridItem = ( { id, title, url } ) => {
    return (
        <div className="card animate_animated animate_fadeIn">
            <img src={ url } alt={ title } />
            <p>{ title }</p>
        </div>
    )
}
