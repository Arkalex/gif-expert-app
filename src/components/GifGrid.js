import React, { useState, useEffect } from 'react'
import { GifGridItem } from './GifGridItem';
import { getGifs } from '../helpers/getGifs';

export const GifGrid = ({ category }) => {

    const [images, setImages] = useState([]);

    useEffect( ()=> {
        getGifs(category)
            .then(imgs => setImages(imgs))
    }, [category])

    

    return (
        <>
            <h3>{category}</h3>
            <div className="card-grid">
            {
                images.map((img) => {
                    return <GifGridItem 
                                key={ img.id }
                                { ...img } 
                            />
                })
            }
            </div>
        </>
    )
}
