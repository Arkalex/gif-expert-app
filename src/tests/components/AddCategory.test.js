import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('Pruebas en <AddCategory />', () => {

    const setCategories = jest.fn();
    let wrapper;

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    })

    test('should show it correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should change input', () => {
        const input = wrapper.find('input');
        const value = 'Hello world';
        input.simulate('change', { target: {value} });

        expect(wrapper.find('p').text().trim()).toBe(value);
    })
    
    test('should submit', () => {
        wrapper.find('form').simulate('submit', {
            preventDefault(){}
        })

        expect(setCategories).not.toHaveBeenCalled()
    })
    
    test('should call setCategories and clean input box', () => {
        const value = 'Hello world';

        //1- Simulate inputChange
        wrapper.find('input').simulate('change', {target: {value}});

        //2- Simulate submit
        wrapper.find('form').simulate('submit', { preventDefault(){} })

        //3- Call setCategories
        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
    
        //4- Input value should be empty string
        expect(wrapper.find('input').prop('value')).toBe('');
    
    })
    
})