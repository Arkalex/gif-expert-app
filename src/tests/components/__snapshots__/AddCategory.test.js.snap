// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> should show it correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
